button-card.js

```diff
@@ -121,118 +127,173 @@
+        else
+          colorValue = config.color;
+      }
+      if (colorValue == 'auto') {
+        if (state) {
+          const color = state.attributes.rgb_color ? `rgb(${state.attributes.rgb_color.join(',')})` : config.default_color;
+          return color;
         } else {
-          if (config.color === 'auto') {
-            color = state.attributes.rgb_color ? `rgb(${state.attributes.rgb_color.join(',')})` : config.default_color;
+          return config.default_color;
+        }
+      } else {
+        if (!colorValue) {
+          if (state) {
+            if (state.state == 'off') {
+              const color = config.color_off;
+              return color;
+            } else {
+              const color = config.default_color;
+              return color;
+            }
+          } else {
+            const color = config.default_color;
+            return color;
           }
-          color = state.state === 'on' ? color : config.color_off;
+        } else {
+          const color = colorValue;
+          return color;
         }
       }
-      return color;
     }
 
-    buildIcon(state, config) {
-      let iconOff = config.icon;
-      if (config.icon == 'attribute') {
+    buildIcon(state, config, configState) {
+      let iconValue = null;
+      if (configState && configState.icon) {
+        iconValue = configState.icon;
+      } else {
+        iconValue = config.icon;
+      }
+      if (iconValue == 'attribute') {
         if (state) {
           const icon = state.attributes.icon;
           return icon;
+        } else {
+          return undefined;
         }
-        return iconOff;
-      }
-      let configState = this.testConfigState(state, config);
-      if (configState && configState.icon) {
-        const icon = configState.icon;
+      } else {
+        const icon = iconValue;
         return icon;
       }
-      return iconOff;
     }
 
-    blankCardColoredHtml(state, config) {
+    buildStyle(state, config, configState) {
+      let cardStyle = '';
+      let styleArray = undefined;
+      if (state) {
+        if (configState && configState.style)
+          styleArray = configState.style;
+        else if (config.style)
+          styleArray = config.style;
+      } else {
+        if (config.style)
+          styleArray = config.style;
+      }
+      if (styleArray) {
+        styleArray.forEach((cssObject) => {
+          const attribute = Object.keys(cssObject)[0];
+          const value = cssObject[attribute];
+          cardStyle += `${attribute}: ${value};\n`;
+        });
+      }
+      return cardStyle;
+    }
+
+    isClickable(state, config) {
+      if (!config.tap_action && state) {
+        switch (state.entity_id.split('.', 2)[0]) {
+          case 'sensor':
+          case 'binary_sensor':
+            return false
+          default:
+            return true
+        }
+      } else {
+        if (config.tap_action && config.tap_action.action == 'none') return false
```

Comment:
add braces
```suggestion
        if (config.tap_action &&
            config.tap_action.action === 'none') {
            return false
        }
```

Reply:
Shouldn't the same sensor/binary_sensor check be here if the action is `toggle`?

Reply:
I changed the way it works a bit, not needed anymore